### PR TITLE
Feat #9334 Add USD to drop button

### LINKED ISSUE #9334: Add price in USD to Drops
We have had this discussion many times about showing prices in USD. We want to be user-friendly. As it's possible that with these drops there may also be new people coming to Polkadot - I would definitely think this is important to have: Price of the drop in USD too. 

Screenshot 2024-02-09 at 10 41 17

Maybe just add it after Mint for 0.5 DOT (approximately 3 USD) ... or something like that. Also should be mentioned in condition I think.

wdyt guys

### PR DESCRIPTION
**Pull request by Haviles04**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #9334 
- [ ] Requires deployment <worker/stick/speck>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [x] I've tested it at ahp/drops/motifs
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=12bcbmuScAcivbqGCpeyRsrPe18WfhotvNywd7ci27CTWHAw)

#### Community participation

- [x] [Are you at Koda Ecosystem Telegram?](https://t.me/koda_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

-Added eager watch to variables to compute amount in USD
-Updated computed value to show USD if available

### CODE REVIEW

**Comment by prury** [comment]
Hello, found a problem: 
Changing the amount of NFTs to be minted does not change the USD value accordingly: 

https://github.com/kodadot/nft-gallery/assets/36627808/e243c08d-ce64-4c74-9e5e-835898a9a915

**Comment by Haviles04** [comment]
> Hello, found a problem: 
> Changing the amount of NFTs to be minted does not change the USD value accordingly: 
> 
> https://github.com/kodadot/nft-gallery/assets/36627808/e243c08d-ce64-4c74-9e5e-835898a9a915
> 
> 

Good call, thanks. I will fix this.

**Comment by Haviles04** [comment]
@prury should be fixed

**Comment by prury** [comment]
> @prury should be fixed

thank you

**Comment by yangwao** [comment]
Apart from somehow wrong rounding I bet

image

image

What you are @Haviles04 using for price?
As quick check on Coingecko shows per DOT different price
image

**Comment by yangwao** [comment]
Thinking outside of scope of this PR, this would need having USD too 

image

File: components/collection/drop/MintButton.vue [hunk 1]
```diff
@@ -50,11 +55,20 @@ const isHolderAndEligible = computed(
     holderOfCollection.value.hasAvailable,
 )
 
+watch(drop, async () => {
+  const tokenPrice = await getApproximatePriceOf(chainSymbol.value)
+  const tokenAmount = calculateBalanceUsdValue(
+    Number(drop.value.price),
+    decimals.value,
+  )
+  priceUsd.value = `${Number(tokenAmount * tokenPrice)}`
```
Comment (Jarsen136) [inline components/collection/drop/MintButton.vue:64]: Why convert it to string?
```suggestion
  priceUsd.value = tokenAmount * tokenPrice
```

**Review by Jarsen136 — COMMENTED** [review]
small stuff

File: components/collection/drop/MintButton.vue (hunk 1 above)
Comment (Haviles04) [reply components/collection/drop/MintButton.vue]: no real reason.. fixed

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 9.57 USD/DOT ~ 3.135 $DOT
🧗 [12bcbmuScAcivbqGCpeyRsrPe18WfhotvNywd7ci27CTWHAw](https://polkadot.subscan.io/account/12bcbmuScAcivbqGCpeyRsrPe18WfhotvNywd7ci27CTWHAw)
🔗 [0x74e4debd5a1ddb4e79a61cb417c23765888dced7f82e516ee00dbc2993e6ec28](https://polkadot.subscan.io/extrinsic/0x74e4debd5a1ddb4e79a61cb417c23765888dced7f82e516ee00dbc2993e6ec28)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by Haviles04** [comment]
> 
> 😍 Perfect, I’ve sent the payout 
> 💵 $30 @ 9.57 USD/DOT ~ 3.135 $DOT
> 🧗 [12bcbmuScAcivbqGCpeyRsrPe18WfhotvNywd7ci27CTWHAw](https://polkadot.subscan.io/account/12bcbmuScAcivbqGCpeyRsrPe18WfhotvNywd7ci27CTWHAw)
> 🔗 [0x74e4debd5a1ddb4e79a61cb417c23765888dced7f82e516ee00dbc2993e6ec28](https://polkadot.subscan.io/extrinsic/0x74e4debd5a1ddb4e79a61cb417c23765888dced7f82e516ee00dbc2993e6ec28)
> 
> 🪅 Let’s grab another issue and get rewarded! 
> 🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

Thank you!

**Pull request merged**